digital_image_processing/filters/local_binary_pattern.py

```diff
@@ -0,0 +1,96 @@
+import cv2
+import numpy as np
+
+
+def get_neighbors_pixel(
+    image: np.ndarray, x_coordinate: int, y_coordinate: int, center: int
+) -> int:
+    """
+    Comparing local neighborhood pixel value with threshold value of centre pixel.
+    Exception is required when neighborhood value of a center pixel value is null.
+    i.e. values present at boundaries.
+
+    :param image: The image we're working with
+    :param x_coordinate: x-coordinate of the  pixel
+    :param y_coordinate: The y coordinate of the pixel
+    :param center: center pixel value
+    :return: The value of the pixel is being returned.
+    """
+    value = 0
+
+    try:
+        if image[x_coordinate][y_coordinate] >= center:
+            value = 1
+    except Exception:
+        pass
+
+    return value
+
+
+def local_binary_value(image: np.ndarray, x_coordinate: int, y_coordinate: int) -> int:
+    """
+    It takes an image, an x and y coordinate, and returns the
+    decimal value of the local binary patternof the pixel
+    at that coordinate
+
+    :param image: the image to be processed
+    :param x_coordinate: x coordinate of the pixel
+    :param y_coordinate: the y coordinate of the pixel
+    :return: The decimal value of the binary value of the pixels
+    around the center pixel.
+    """
+    binary_values = []
+    center = image[x_coordinate][y_coordinate]
+    powers = [1, 2, 4, 8, 16, 32, 64, 128]
+
+    # Starting from top right,assigning value to pixels clockwise
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate + 1, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate, y_coordinate + 1, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate + 1, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate - 1, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate, y_coordinate - 1, center)
+    )
+    binary_values.append(
+        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate - 1, center)
+    )
```

Comment:
Readability and performance...
```suggestion
    center = image[x_coordinate][y_coordinate]
    powers = [1, 2, 4, 8, 16, 32, 64, 128]

    # Starting from the top right, assigning value to pixels clockwise
    binary_values = [
        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate + 1, center),
        get_neighbors_pixel(image, x_coordinate, y_coordinate + 1, center),
        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate, center),
        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate + 1, center),
        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate, center),
        get_neighbors_pixel(image, x_coordinate + 1, y_coordinate - 1, center),
        get_neighbors_pixel(image, x_coordinate, y_coordinate - 1, center),
        get_neighbors_pixel(image, x_coordinate - 1, y_coordinate - 1, center),
    ]
```